### DISCUSSION TITLE
Type Error: response2.headers.all is not a function

### DISCUSSION THREAD

**Discussion by markgeorgebg** [mswjs/msw#1884] [Q&A] [answered]
Hi

I have started getting the error "Type Error: response2.headers.all is not a function" on all my mocks. They were working fine, but now they are all broken because of this. I have the exact same setup on another repo and it is working and has been working for a long time. Has anyone had this issue before and found a resolution? 

I am running an older version of MSW (1.2.1) but am not in a position to upgrade at this time. 

Any help appreciated. 
Thanks

**Comment by harrinsonmb** [comment, marked as answer]
Hello @markgeorgebg,

I was having the same problem and noticed that the .all() method was removed from headers-polyfill [since version 3.3.0](https://github.com/mswjs/headers-polyfill/releases/tag/v3.3.0 ). So I decided to install an older version as a development dependency and it seems to work fine now.

`npm i headers-polyfill@3.2.5 -D`

I hope it works for you too.

Greetings!

**Reply by markgeorgebg** [reply]
Hey - this worked, thanks for your help - really helped me out of a spot!

**Comment by krzotki** [comment]
I also had this issue with configuring new repo, keeping the same setup. In my case the issue was with `yarn.lock` file - it was generated from scratch after installing deps, but it should've also been copied from the old repo.

**Comment by daniellizik** [comment]
i saw `response3.headers.all is not a function` but the polyfill did the trick, thanks!

node v16.4.2 (yeah i know...)
```ts
axios: 1.7.4
msw:  1.1.0
vitest: 0.29.7
vite: 4.5.5

```

**Comment by GabrielCastilhoV** [comment]
### ✅ Fix: Change `msw` from `"1.2.2"` to `"^1.2.2"`

If you're facing issues with MSW, a simple fix that worked for me was updating the version in `package.json` 

from:

```json
"msw": "1.2.2"
````

to:

```json
"msw": "^1.2.2"
```

Using a fixed version (`"1.2.2"`) locks you to that exact release, which may include bugs. Adding the caret (`^`) allows patch/minor updates (e.g., `1.2.3`, `1.3.x`) that include important fixes — and that alone resolved the issue in my case.
